src/transformers/image_transforms.py

```diff
@@ -200,11 +203,12 @@ def to_pil_image(
     elif not isinstance(image, np.ndarray):
         raise ValueError("Input image type not supported: {}".format(type(image)))
 
-    # If the channel has been moved to first dim, we put it back at the end.
-    image = to_channel_dimension_format(image, ChannelDimension.LAST, input_data_format)
+    # If the channel has been moved to first dim, we put it back at the end if not 2d image
+    if input_data_format != ChannelDimension.NONE and image.shape != 2:
+        image = to_channel_dimension_format(image, ChannelDimension.LAST, input_data_format)
 
-    # If there is a single channel, we squeeze it, as otherwise PIL can't handle it.
-    image = np.squeeze(image, axis=-1) if image.shape[-1] == 1 else image
+        # If there is a single channel, we squeeze it, as otherwise PIL can't handle it.
+        image = np.squeeze(image, axis=-1) if image.shape[-1] == 1 else image
 
```

Comment:
Probably, we will not need this modification if `to_channel_dimension_format` will work with NONE format